src/models/logger.ts

```diff
@@ -0,0 +1,37 @@
+export class Logger {
```

Comment:
Why do we need this class at all though?  I think we can implement a plotLogger as a part of log, which in turn trigger a relevant LogAction directly.